### PR TITLE
docs: batcher max fee priority queue & finalization algorithm

### PR DESCRIPTION
**Pull request by entropidelic**
This PR adds priority queue and batch finalization algorithm explanations to batcher documentation.

### CODE REVIEW

File: docs/2_architecture/components/1_batcher.md [hunk 1]
```diff
@@ -11,3 +11,51 @@ However, each proof has a cost of verification, so each batch must contain some
...
+
+When a user submits a proof for the first time in the batch, its max fee is cached and set as the user min fee for the rest of the proofs to be included in the batch.
+If a later message with a higher max fee is received, the message is rejected and not included in the queue, while if a message with a lower max fee is received,
+the message is included in the queue and the user min fee is updated to that value. In summary, **no messages with a higher max fee than this user's min fee will be included**.
+
+In case a message has a max fee that is too low - making it to be stuck in the batcher's *mempool*,  the message can be re-sent and be replaced with a higher fee.
+To this end, a validation is done first. We check that when the max fee for the message with that nonce is updated, there is no message with a lower nonce and a lower max fee too, because this would lead to the problem
+of messages with higher nonce processed earlier than messages with higher nonce, as discussed earlier.
+As an example, consider all these messages in a batch from the same address:
```
Comment (PatStiles) [inline docs/2_architecture/components/1_batcher.md]: This is hard to understand

File: docs/2_architecture/components/1_batcher.md [hunk 2]
```diff
@@ -11,3 +11,105 @@ However, each proof has a cost of verification, so each batch must contain some
...
+
+```
+priority_queue = [(E, 74), (D, 75), (C, 90), (B, 95), (A, 100)]
+resulting_priority_queue = []
+max_batch_size = 1000 # Defined constant
+```
+
+On the first iteration, the proof with the smallest **max fee** is taken and the **fee per proof** is calculated
+
+```
```
Comment (PatStiles) [inline docs/2_architecture/components/1_batcher.md:85]: Add priority queue state description to this example section as well as seen above and below.

**Review by PatStiles — APPROVED** [review]
One nit!

**Review by uri-99 — APPROVED** [review]

File: docs/2_architecture/components/1_batcher.md (hunk 2 above)
Comment (JulianVentura) [reply docs/2_architecture/components/1_batcher.md:85]: Done

**Review by MarcosNicolau — APPROVED** [review]
LGTM

**Pull request merged**